htlcswitch/switch_test.go

```diff
@@ -4318,22 +4318,12 @@ func TestSwitchDustForwarding(t *testing.T) {
 		OnionBlob:   blob,
 	}
 
```

Comment:
I tried to understand this testcase but tbh, I don't really understand it. In the description it says that we are testing the treshold here however I think what we are really testing is the channel contraint of the max_inflight_htlcs we are allowed to put onto a channel. So my question is what do we really want to test here.

When sending dust (`sendDustHtlcs`) we are putting 50 htlcs onto the channel in each direction, the 51 one fails:

```
2024-10-31 10:28:31.426 [WRN] HSWC: ChannelLink(bcfe226d21c9936ee578df4afb65835abb9790392e45bbaf0c245394430838f8:0): Unable to handle downstream add HTLC: commitment transaction exceed max htlc number
2024-10-31 10:28:31.426 [DBG] HSWC: Storing result for attemptID=50
2024-10-31 10:28:31.439 [DBG] HSWC: Tearing down circuit with FAIL pkt, removing circuit=(Chan ID=0:0:0, HTLC ID=50) with keystone=(Chan ID=0:0:0, HTLC ID=0)
```

And all the followup tests just try to add another htlc to the link but its already blocked, so I think this testcase might be a relict of the former dust-treshold implementation, which now changed and is by default 500_000 sats ?

Reply:
the main test I think is in `checkAlmostDust` or `assertAlmostDust`, so this PR focuses on removing the test flakes, but yeah agree we should start making the tests here better since it gives flakes quite often.

Reply:
decreased the fee-exposure and made sure we hit the error of being overexposed and not hitting the error that we hit the channel contraint.